### PR TITLE
Story/vspc-56

### PR DESCRIPTION
**Pull request by ashmichheda**

### CODE REVIEW

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditSequenceController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import edu.asu.diging.vspace.core.model.ISequence;
+import edu.asu.diging.vspace.core.model.impl.Sequence;
+import edu.asu.diging.vspace.core.services.ISequenceManager;
+
+@Controller
+public class EditSequenceController {
+
+	
+	@Autowired
+	ISequenceManager sequenceManager;
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditSequenceController.java]: should be private

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditSequenceController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    @RequestMapping(value = "/staff/module/{moduleId}/sequence/{sequenceId}/edit/title", method = RequestMethod.POST)
+    public ResponseEntity<String> saveTitle(@RequestParam("title") String title,
+            @PathVariable("moduleId") String moduleId, @PathVariable("sequenceId") String sequenceId) {
+    	ISequence sequence = sequenceManager.getSequence(sequenceId);
+    	sequence.setName(title);
+        sequenceManager.updateSequence((Sequence) sequence);
+        return new ResponseEntity<String>(HttpStatus.OK);
+    }
+}
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditSequenceController.java]: indention is all over the place

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 3]
```diff
@@ -10,13 +10,168 @@
...
+        
+    });
+   // ------- Edit/Save Description ends --------
+});
+
+
+
+</script>
+
+<!--  <h1>Sequence: ${sequence.name}</h1> -->
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why is this commented out? if it's not needed anymore, remove.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 4]
```diff
@@ -10,13 +10,168 @@
...
+        processData : false,
+        data: formData,
+        enctype: 'multipart/form-data',
+        success: function(data) {
+            // replace text box with new description
+            $("#submitDescription").hide()
+            $("#cancelEditDescription").hide()
+            $("#editDescription").show()
+            var val = $("#newDescription").val();
+            console.log("Value of new description: "+val)
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: remove

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 5]
```diff
@@ -10,13 +10,168 @@
...
+            data: formData,
+            enctype: 'multipart/form-data',
+            success: function(data) {
+                // replace text box with new description
+                $("#submitTitle").hide()
+                $("#cancelEditTitle").hide();
+                $("#editTitle").show()
+                var val = $("#newTitle").val();
+                $("#newTitle").closest('div').remove();
+                $("#title").text("Silde: " + val)
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why does this say "Slide"?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 6]
```diff
@@ -10,13 +10,168 @@
...
+                $('.error').append(alert);
+            }
+        });
+      });
+   $("#cancelEditTitle").click(function(){
+        $("#submitTitle").hide()
+        $("#editTitle").show()
+        $("#cancelEditTitle").hide()
+        $("#newTitle").closest('div').remove();
+        $("#title").text("Silde: " + getTitleText)
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why does this say "slide"?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 7]
```diff
@@ -10,13 +10,168 @@
...
+    >Cancel</button>
+</div>
 <div class="alert alert-light" role="alert">
   Created on <span class="date">${sequence.creationDate}</span> by ${sequence.createdBy}.<br>
   Modified on <span class="date">${sequence.modificationDate}</span> by ${sequence.modifiedBy}.
 </div>
-<h5>Description:</h5>
-<p>${sequence.description}</p>
+ <!-- <h5>Description:</h5>
+<p>${sequence.description}</p>  -->
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: remove if not needed

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SequenceManager.java [hunk 8]
```diff
@@ -52,4 +52,9 @@ public ISequence getSequence(String sequenceId) {
         }
         return null;
     }
+
+    @Override
+    public void updateSequence(Sequence sequence) {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SequenceManager.java]: the interface should be used here. otherwise why would you cast in the next line?

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditSequenceController.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Autowired
+    private ISequenceManager sequenceManager;
+
+    @RequestMapping(value = "/staff/module/{moduleId}/sequence/{sequenceId}/edit/description", method = RequestMethod.POST)
+    public ResponseEntity<String> saveDescription(@RequestParam("description") String description,
+            @PathVariable("moduleId") String moduleId, @PathVariable("sequenceId") String sequenceId) {
+        ISequence sequence = sequenceManager.getSequence(sequenceId);
+        sequence.setDescription(description);
+        sequenceManager.updateSequence((Sequence) sequence);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditSequenceController.java]: should not be cast here

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditSequenceController.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        return new ResponseEntity<String>(HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/staff/module/{moduleId}/sequence/{sequenceId}/edit/title", method = RequestMethod.POST)
+    public ResponseEntity<String> saveTitle(@RequestParam("title") String title,
+            @PathVariable("moduleId") String moduleId, @PathVariable("sequenceId") String sequenceId) {
+        ISequence sequence = sequenceManager.getSequence(sequenceId);
+        sequence.setName(title);
+        sequenceManager.updateSequence((Sequence) sequence);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditSequenceController.java]: same here

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
When I edit the description, it changes it. but when I edit it again (without reloading the page) it still shows the old description.
- edit description
- click save
- click edit icon again
-> previous description (edit in step 1) is still in input field

**Comment by jdamerow** [comment]
same for title field

**Comment by jdamerow** [comment]
Jenkins, test this please.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 11]
```diff
@@ -1,23 +1,184 @@
...
+    
+    
+    
+    //------- edit title of Sequence starts --------
+    
+    $("#submitTitle").hide();
+    $("#cancelEditTitle").hide();
+    
+    var getTitleText = $("#title").text().split(": ")[1]
+    console.log("Value of title: "+getTitleText);
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: remove

**Comment by jdamerow** [comment]
Nope, still the same problem. Also there is something weird with the title. It seems to get tabs appended.

**Comment by jdamerow** [comment]
cancel now shows the new entered (cancelled) text, not the actual text.

**Comment by ashmichheda** [comment]
While editing sequence Title, in order to retrieve the title value i have kept the value displayed while edit. Please let me know if this is okay, I will do the same change for **description**.
image

**Comment by jdamerow** [comment]
That'll become very unwieldy for longer titles and especially for descriptions. What's the reasoning for that. It shouldn't be necessary to keep the old title/description around.

**Comment by ashmichheda** [comment]
Fixed the the title and description of sequence. 
Fix - Saved the original values of title and description using value() function for retrieving it when clicked 'cancel' button.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 12]
```diff
@@ -1,23 +1,200 @@
...
+$(document).ready(function() { 
+ 
+	
+    //-------- edit contentblock description Sequence starts --------
+    $("#submitDescription").hide()
+    $("#cancelEditDescription").hide()	
+    
+    $("#editDescription").click(function() {
+    	var description = $("#description").text()
+    	document.getElementById("description").value = description;
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: we're using jquery. let's not mix jquery and plain js unless we have to.

Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: also, why do you set the value field of the description element?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 13]
```diff
@@ -1,23 +1,200 @@
...
+        processData : false,
+        data: formData,
+        enctype: 'multipart/form-data',
+        success: function(data) {
+            // replace text box with new description
+            $("#submitDescription").hide()
+            $("#cancelEditDescription").hide()
+            $("#editDescription").show()
+            var val = $("#newDescription").val();
+            $('<p id="description"style="margin-top: .5rem; margin-bottom: .5rem;">val</p>').insertBefore( "#newDescription" );
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why are you inserting a new element with id "description"? the old one is only hidden, isn't it?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 14]
```diff
@@ -1,23 +1,200 @@
...
+   });
+    
+    // ---- Cancel Edited Description ----
+    
+    $("#cancelEditDescription").click(function(){
+        $("#submitDescription").hide()
+        $("#editDescription").show()
+        $("#cancelEditDescription").hide()
+        var description = document.getElementById("description").value
+        $('<p id="description" style="margin-top: .5rem; margin-bottom: .5rem;">description</p>').insertBefore( "#newDescription" );
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: same here, what's with the old element with that id?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 15]
```diff
@@ -1,23 +1,200 @@
...
+    // ------- Edit/Save/Cancel Description ends --------
+    
+    
+    
+    //------- edit title of Sequence starts --------
+    
+    $("#submitTitle").hide();
+    $("#cancelEditTitle").hide();
+    
+    	$("#editTitle").click(function() {
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: indentation

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 16]
```diff
@@ -1,23 +1,200 @@
...
+
+    	var getTitleText = $("#title").text().split(": ")[1].trim()
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: ugly, use data attributes or another cleaner way to get the title

Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: also, bad variable name. get... sounds like a method

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 17]
```diff
@@ -1,23 +1,200 @@
...
+        $('<div class="col-4"><input id="newTitle" class="form-control" type="text"></div>').insertAfter( "#title" );
+    	document.getElementById("title").value = getTitleText;
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: same as above, don't mix jquery and plain js

Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: and why do you set the value attribute?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 18]
```diff
@@ -1,23 +1,200 @@
...
+            processData : false,
+            data: formData,
+            enctype: 'multipart/form-data',
+            success: function(data) {
+                // replace text box with new description
+                $("#submitTitle").hide()
+                $("#cancelEditTitle").hide();
+                $("#editTitle").show()
+                var val = $("#newTitle").val();
+                $("#newTitle").closest('div').remove();
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why don't you give the div an id and remove it by id? much less error prone, because what if we need some extra layouting or something and need to add another div around the input field?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 19]
```diff
@@ -1,23 +1,200 @@
...
+      });
+    
+    
+   // ------  Edit Title Sequence ------
+   
+   $("#cancelEditTitle").click(function(){
+        $("#submitTitle").hide()
+        $("#editTitle").show()
+        $("#cancelEditTitle").hide()
+       	var value = document.getElementById("title").value
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: indentation

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 20]
```diff
@@ -1,23 +1,200 @@
...
+        $("#newTitle").closest('div').remove();
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: same as above

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 21]
```diff
@@ -1,23 +1,200 @@
...
-<p>${sequence.description}</p>
-    
+
+
+
+<!-- description -->
+<div style="margin-left: .1%;" class="row align-items-center">
+	<h4 style="margin-bottom: 0px;">
+		Description:
+		</h5>
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: you are opening an h4 but closing an h5

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 12 above)
Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: In the code, value of the description is getting removed. Hence, I stored it in value attribute so that it can be retrieved later. Otherwise, it would be difficult to fetch original value.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 13 above)
Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: I have referred to the functionality implemented for Slides.jsp for it's title and description and have incorporated it for sequence.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 12 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: I don't think it's proper HTML to give p elements a value attribute. Use data attributes instead.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 13 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: comment still applied. if you end up with multiple elements with id "description" something is wrong.

Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: okay, I will use different id names, I tried with the one that is hidden, but it's not working.

Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: it's not about using a different id but about reusing the existing element.

**Comment by jdamerow** [comment]
what about my other comments about val attributes?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 22]
```diff
@@ -1,23 +1,208 @@
...
+        data: formData,
+        enctype: 'multipart/form-data',
+        success: function(data) {
+            // replace text box with new description
+            $("#submitDescription").hide()
+            $("#cancelEditDescription").hide()
+            $("#editDescription").show()
+            var val = $("#newDescription").val();
+            $("#editDescription").show()
+            $('<p id="description" style="margin-top: .5rem; margin-bottom: .5rem;">val</p>').insertBefore( "#newDescription" );
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: you never deleted the old description element, did you? wouldn't you end up with multiple? why are you not using the previous one?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 23]
```diff
@@ -1,23 +1,208 @@
...
+   });
+    
+    // ---- Cancel Edited Description ----
+    
+    $("#cancelEditDescription").click(function(){
+        $("#submitDescription").hide()
+        $("#editDescription").show()
+        $("#cancelEditDescription").hide()
+        var description = $("#description").data('value')
+        $('<p id="editedDescription" style="margin-top: .5rem; margin-bottom: .5rem;">'+description+'</p>').insertBefore( "#newDescription" );
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why are you creating this if you delete in line 90?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 24]
```diff
@@ -1,23 +1,208 @@
...
+    //------- edit title of Sequence starts --------
+    
+    $("#submitTitle").hide();
+    $("#cancelEditTitle").hide();
+    
+    $("#editTitle").click(function() {
+    	// Using data attribute
+    	var sequenceTitle = $("#title").data('value')  // gets value without using trim function
+        $('<div class="col-4" id = "editSequenceTitle"><input id="newTitle" class="form-control" type="text"></div>').insertAfter( "#title" );
+    	$("#title").val(sequenceTitle)
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why are you setting the value here?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 25]
```diff
@@ -1,23 +1,207 @@
...
+        enctype: 'multipart/form-data',
+        success: function(data) {
+            // replace text box with new description
+            $("#submitDescription").hide()
+            $("#cancelEditDescription").hide()
+            $("#editDescription").show()
+            var val = $("#newDescription").val();
+            $("#editDescription").show()
+            $("#description").show() 
+            $('#description').insertBefore( "#newDescription" );
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why are you inserted newDescription here?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 26]
```diff
@@ -1,23 +1,207 @@
...
+            $("#description").text(val);
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: first set the text, then show it. otherwise the user see's the flicker of changing the description.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 27]
```diff
@@ -1,23 +1,207 @@
...
+            $("#description").data('value',val);
+            $("#newDescription").hide()
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: every time the user edits the description, you create a new textarea and add it to the html. that would result in many textareas that are not used, wouldn't it?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 28]
```diff
@@ -1,23 +1,207 @@
...
+    // ---- Cancel Edited Description ----
+    
+    $("#cancelEditDescription").click(function(){
+        $("#submitDescription").hide()
+        $("#editDescription").show()
+        $("#cancelEditDescription").hide()
+        var description = $("#description").data('value')
+        $('description').insertBefore( "#newDescription" );
+        $("#newDescription").remove()
+        $("#description").text(description);
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: the description shouldn't have changed, should it? so no need to set the text?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 29]
```diff
@@ -1,23 +1,207 @@
...
+   });
+    
+    // ---- Cancel Edited Description ----
+    
+    $("#cancelEditDescription").click(function(){
+        $("#submitDescription").hide()
+        $("#editDescription").show()
+        $("#cancelEditDescription").hide()
+        var description = $("#description").data('value')
+        $('description').insertBefore( "#newDescription" );
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: see above

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 24 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: ?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 27 above)
Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: Previously, the code was creating new description tag, while editing each time. Hence, as you suggested in the previous pull request to use the original description tag. I have reused the same. It's not creating a new one, just using the "description" tag initially created.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 28 above)
Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: Here the value changes, if I edit any value without saving it. Hence, I had to set the text tag.
I found out while testing it.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 27 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: line 32 creates a new textarea, doesn't it?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 28 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: if you never change the text of the div description, how can it change?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 27 above)
Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: Yes, because while editing we need a text-area to edit. As there is no already created text-area for description. Hence we are creating it. I will close it on submitting, to avoid multiple creations.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 28 above)
Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: Yes, there is no need to set the text. I will remove it.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 25 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: if you delete newDescription in line 68, why inserting it here?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 29 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: see above

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 24 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: ?

Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: please clarify/fix

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 30]
```diff
@@ -1,23 +1,206 @@
...
+      });
+    
+    
+   // ------  Edit Title Sequence ------
+   
+   $("#cancelEditTitle").click(function(){
+        $("#submitTitle").hide()
+        $("#editTitle").show()
+        $("#cancelEditTitle").hide()
+        var sequenceTitle = $("#title").val()
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: there is not value attribute for h2

Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: if you delete newDescription in line 68, why inserting it here?
-- I am not inserting new Description again. I am just added original description tag before "newDescription" using insertBefore and then deleting "newDescription"

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 25 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why? description is already in your html, why would you insert it again?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 29 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: see above (also, this doesn't refer to anything 'description' would refer to a tag, '#description' refers to an element with id description, so if this works, you can delete this line).

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 24 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: FIX THIS

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 30 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why adding the original description again? it's already in your html.

Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: h2 does not have a value attribute

**Comment by ashmichheda** [comment]
var sequenceTitle = $("#title").data('value')  // gets value without using trim function
        $('<div class="col-4" id = "editSequenceTitle"><input id="newTitle" class="form-control" type="text"></div>').insertAfter( "#title" );
    	$("#title").val(sequenceTitle)


For the above, what needs to be fixed?

**Comment by ashmichheda** [comment]
var sequenceTitle = $("#title").data('value')  // gets value without using trim function
        $('<div class="col-4" id = "editSequenceTitle"><input id="newTitle" class="form-control" type="text"></div>').insertAfter( "#title" );
    	$("#title").val(sequenceTitle)

**Comment by ashmichheda** [comment]
When editTitle function is called, it sets the sequence value to "Sequence:", due to which I cannot retrieve the value.

$("#editTitle").click(function() {
    	// Using data attribute
    	var sequenceTitle = $("#title").data('value')  // gets value without using trim function
        $('<div class="col-4" id = "editSequenceTitle"><input id="newTitle" class="form-control" type="text"></div>').insertAfter( "#title" );
    	$("#title").val(sequenceTitle)
    	**$('#title').text('Sequence: ')**
        $("#newTitle").val(sequenceTitle)
        $("#editTitle").hide()
        $("#submitTitle").show()
        $("#cancelEditTitle").show()
        
    });

The line that is bold.
So, how can I added a data-attribute to set the value.
Is there any other way, I can retrieve it.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 25 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why is this here? what's the point of it? both #description and @newDescription are already in the dom tree.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 29 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: same here

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 24 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: really?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 31]
```diff
@@ -1,23 +1,205 @@
...
+
+</script>
+
+
+<!-- title -->
+<div class="row align-items-center">
+	<h2 id="title" style="margin-bottom: 0%; margin-left: 1%;" data-value="${sequence.name}">
+		Sequence: ${sequence.name}
+		</h2>
+		<p id="titleName" style="margin-top: .5rem; font-size:35px; margin-bottom: .5rem;" data-value="${sequence.name}"></p>
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why is this here?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 32]
```diff
@@ -1,23 +1,205 @@
...
+    $("#submitTitle").hide();
+    $("#cancelEditTitle").hide();
+    
+    $("#editTitle").click(function() {
+    	// Using data attribute
+    	var sequenceTitle = $("#titleName").data('value')  // gets value without using trim function
+        $('<div class="col-4" id = "editSequenceTitle"><input id="newTitle" class="form-control" type="text"></div>').insertAfter( "#title" );
+    	$("#title").val(sequenceTitle)
+    	$('#title').text('Sequence: ')
+        $("#newTitle").val(sequenceTitle)
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: why is there a title and a newtitle? what's that for? And since they are h2 and p, why do you use val() here?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 31 above)
Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: I used the p tag, since h2 tag cannot have val attribute. So in order to retrieve the Sequence name, I used p tag and got the value using data-attribute

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 32 above)
Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: Since title is <h2> tag, while editing title, newTitle of the input form is added so that user can add any new title.

Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: Yes, I will remove the val()

Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: I'm still confused why you don't use the h2 tag?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 31 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: data attributes should be valid on any tag as far as I know

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp (hunk 32 above)
Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: Because, in the code,
When editTitle function is called, it sets the sequence value to "Sequence:", due to which I cannot retrieve the value from the h2 tag.

$("#editTitle").click(function() {
// Using data attribute
var sequenceTitle = $("#title").data('value') // gets value without using trim function
$('

').insertAfter( "#title" );
$("#title").val(sequenceTitle)
**$('#title').text('Sequence: ')**


Hence, I have added p tag to get the value back.

Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: I can only retrieve "Sequence :" as value, and not the actual value, e.g. **Sequence value: abcd**

Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: I don't get it, is it a display issue? Why don't you do
```
$('#title').data('previous-title', sequenceTitle);
```
when start editing and then
```
var sequenceTitle = $("#title").data('previous-title')  // gets value
$("#title").text("Sequence: "+sequenceTitle)
```
?

Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: I can't use data attributes in title, since it's an H2 tag right?

Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: Hence, I am using a p tag with data attributes

Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: https://www.w3schools.com/tags/att_data-.asp

Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: Okay, I will use data-attributes in h2 tag and remove the p tag.

**Comment by jdamerow** [comment]
👍
---
Julia Damerow, PhD
Scientific Software Engineer
Global Biosocial Complexity Initiative
Digital Innovation Group
Arizona State University
email: jdamerow@asu.edu <>
web: http://diging.asu.edu, https://complexity.asu.edu/

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp [hunk 33]
```diff
@@ -1,23 +1,201 @@
...
+        
+   });
+    
+    // ---- Cancel Edited Description ----
+    
+    $("#cancelEditDescription").click(function(){
+        $("#submitDescription").hide()
+        $("#editDescription").show()
+        $("#cancelEditDescription").hide()
+        var description = $("#description").data('value')
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: where is that being used?

Comment (ashmichheda) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/sequence.jsp]: Yes, its not being used. Will remove that.

**Comment by jdamerow** [comment]
Whoohoo! ;)

**Comment by ashmichheda** [comment]
:)
